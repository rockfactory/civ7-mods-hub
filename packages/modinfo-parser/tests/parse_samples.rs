use modinfo_parser::Mod;

#[test]
fn parse_samples() {
    insta::glob!("samples/*.modinfo", |path| {
        let text = std::fs::read_to_string(path).unwrap();
        let parsed = Mod::parse(text.as_bytes()).unwrap();
        insta::with_settings!({ sort_maps => true }, {
            insta::assert_debug_snapshot!(parsed);
            let serialized = parsed.to_string().unwrap();
            let reparsed = Mod::parse(serialized.as_bytes()).unwrap();
            assert_eq!(parsed, reparsed);
        });
    });
}
