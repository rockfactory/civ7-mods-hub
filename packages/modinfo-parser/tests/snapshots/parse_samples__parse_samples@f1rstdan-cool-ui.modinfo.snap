---
source: packages/modinfo-parser/tests/parse_samples.rs
expression: parsed
input_file: packages/modinfo-parser/tests/samples/f1rstdan-cool-ui.modinfo
---
Mod {
    id: "f1rstdan-cool-ui",
    version: Some(
        "1",
    ),
    properties: Properties {
        name: Some(
            "LOC_MOD_F1RSTDAN_COOL_UI_NAME",
        ),
        description: Some(
            "LOC_MOD_F1RSTDAN_COOL_UI_DESCRIPTION",
        ),
        authors: Some(
            "F1rstDan",
        ),
        package: Some(
            "Mod",
        ),
        affects_saved_games: false,
        special_thanks: None,
        version: Some(
            "1.7.4",
        ),
        compatibility: None,
        url: Some(
            "https://forums.civfanatics.com/resources/31961/",
        ),
    },
    dependencies: [
        Dependency {
            id: "base-standard",
            title: Some(
                "LOC_MODULE_BASE_STANDARD_NAME",
            ),
        },
    ],
    localized_text: [
        "text/en_us/ModInfoText.xml",
        "text/zh_Hans_CN/ModInfoText.xml",
        "text/ko_KR/ModInfoText.xml",
        "text/ru_RU/ModInfoText.xml",
    ],
    action_criteria: [
        ActionCriterion {
            id: "always",
            always_met: true,
            age_in_use: None,
            mod_in_use: None,
        },
    ],
    action_groups: [
        ActionGroup {
            id: "f1rstdan-menu",
            scope: Some(
                "shell",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    1111,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [],
                    ui_scripts: [
                        ActionItem {
                            path: "ui/options/f1rstdan-cool-ui-options.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/options/mod-options-decorator.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [
                        ActionItem {
                            path: "text/en_us/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/zh_Hans_CN/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/ko_KR/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/ru_RU/InGameText.xml",
                            locale: None,
                        },
                    ],
                },
            ],
        },
        ActionGroup {
            id: "f1rstdan-cool-ui-city",
            scope: Some(
                "game",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    1111,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [
                        ActionItem {
                            path: "ui/production-chooser/panel-production-chooser.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/production-chooser/production-chooser-unique-quarter.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/production-chooser/dan-quick-buy-item.js",
                            locale: None,
                        },
                    ],
                    ui_scripts: [
                        ActionItem {
                            path: "ui/dan-city-production/dan-item-layout-decorator.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [
                        ActionItem {
                            path: "text/en_us/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/zh_Hans_CN/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/ko_KR/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/ru_RU/InGameText.xml",
                            locale: None,
                        },
                    ],
                },
            ],
        },
        ActionGroup {
            id: "f1rstdan-cool-ui-city-yields",
            scope: Some(
                "game",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    1111,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [
                        ActionItem {
                            path: "textures/F1dan_city_population.png",
                            locale: None,
                        },
                        ActionItem {
                            path: "textures/F1dan_city_connectivity.png",
                            locale: None,
                        },
                    ],
                    ui_scripts: [
                        ActionItem {
                            path: "ui/dan-city-yields/dan-city-yields-decorator.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/dan-city-yields/dan-city-yields-tooltip.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [],
                },
            ],
        },
    ],
}
